{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7d98be00",
   "metadata": {},
   "source": [
    "# Package installations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b56a206f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "import pandas as pd\n",
    "import os\n",
    "import requests\n",
    "import subprocess\n",
    "import sys\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ab2cfde1",
   "metadata": {},
   "outputs": [],
   "source": [
    "bik = pd.read_csv('third_dataset.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "49de811d",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: gpt-2-simple in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (0.8.1)\n",
      "Requirement already satisfied: tensorflow>=2.5.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from gpt-2-simple) (2.13.0)\n",
      "Requirement already satisfied: regex in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from gpt-2-simple) (2023.6.3)\n",
      "Requirement already satisfied: requests in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from gpt-2-simple) (2.31.0)\n",
      "Requirement already satisfied: tqdm in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from gpt-2-simple) (4.65.0)\n",
      "Requirement already satisfied: numpy in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from gpt-2-simple) (1.24.3)\n",
      "Requirement already satisfied: toposort in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from gpt-2-simple) (1.10)\n",
      "Requirement already satisfied: absl-py>=1.0.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (1.4.0)\n",
      "Requirement already satisfied: astunparse>=1.6.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (1.6.3)\n",
      "Requirement already satisfied: flatbuffers>=23.1.21 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (23.5.26)\n",
      "Requirement already satisfied: gast<=0.4.0,>=0.2.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (0.4.0)\n",
      "Requirement already satisfied: google-pasta>=0.1.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (0.2.0)\n",
      "Requirement already satisfied: h5py>=2.9.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (3.9.0)\n",
      "Requirement already satisfied: libclang>=13.0.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (16.0.6)\n",
      "Requirement already satisfied: opt-einsum>=2.3.2 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (3.3.0)\n",
      "Requirement already satisfied: packaging in /Users/alberli/Library/Python/3.11/lib/python/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (23.0)\n",
      "Requirement already satisfied: protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (4.23.4)\n",
      "Requirement already satisfied: setuptools in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (65.5.0)\n",
      "Requirement already satisfied: six>=1.12.0 in /Users/alberli/Library/Python/3.11/lib/python/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (1.16.0)\n",
      "Requirement already satisfied: termcolor>=1.1.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (2.3.0)\n",
      "Requirement already satisfied: typing-extensions<4.6.0,>=3.6.6 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (4.5.0)\n",
      "Requirement already satisfied: wrapt>=1.11.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (1.15.0)\n",
      "Requirement already satisfied: grpcio<2.0,>=1.24.3 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (1.56.2)\n",
      "Requirement already satisfied: tensorboard<2.14,>=2.13 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (2.13.0)\n",
      "Requirement already satisfied: tensorflow-estimator<2.14,>=2.13.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (2.13.0)\n",
      "Requirement already satisfied: keras<2.14,>=2.13.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (2.13.1)\n",
      "Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorflow>=2.5.1->gpt-2-simple) (0.32.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from requests->gpt-2-simple) (3.2.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/alberli/Library/Python/3.11/lib/python/site-packages (from requests->gpt-2-simple) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from requests->gpt-2-simple) (1.26.16)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from requests->gpt-2-simple) (2023.7.22)\n",
      "Requirement already satisfied: wheel<1.0,>=0.23.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from astunparse>=1.6.0->tensorflow>=2.5.1->gpt-2-simple) (0.41.0)\n",
      "Requirement already satisfied: google-auth<3,>=1.6.3 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (2.22.0)\n",
      "Requirement already satisfied: google-auth-oauthlib<1.1,>=0.5 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (1.0.0)\n",
      "Requirement already satisfied: markdown>=2.6.8 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (3.4.4)\n",
      "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (0.7.1)\n",
      "Requirement already satisfied: werkzeug>=1.0.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (2.3.6)\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-auth<3,>=1.6.3->tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (5.3.1)\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-auth<3,>=1.6.3->tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (0.3.0)\n",
      "Requirement already satisfied: rsa<5,>=3.1.4 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-auth<3,>=1.6.3->tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (4.9)\n",
      "Requirement already satisfied: requests-oauthlib>=0.7.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-auth-oauthlib<1.1,>=0.5->tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (1.3.1)\n",
      "Requirement already satisfied: MarkupSafe>=2.1.1 in /Users/alberli/Library/Python/3.11/lib/python/site-packages (from werkzeug>=1.0.1->tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (2.1.2)\n",
      "Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (0.5.0)\n",
      "Requirement already satisfied: oauthlib>=3.0.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<1.1,>=0.5->tensorboard<2.14,>=2.13->tensorflow>=2.5.1->gpt-2-simple) (3.2.2)\n"
     ]
    }
   ],
   "source": [
    "# ! pip3 install gpt-2-simple"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9a79bda",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !python3 -m pip install gpt-2-simple"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f96f18a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ! pip3 install --upgrade pip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4bdd8b0b",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting tika\n",
      "  Using cached tika-2.6.0-py3-none-any.whl\n",
      "Requirement already satisfied: setuptools in /Users/alberli/opt/anaconda3/envs/pythoneleven/lib/python3.11/site-packages (from tika) (68.0.0)\n",
      "Requirement already satisfied: requests in /Users/alberli/opt/anaconda3/envs/pythoneleven/lib/python3.11/site-packages (from tika) (2.31.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /Users/alberli/opt/anaconda3/envs/pythoneleven/lib/python3.11/site-packages (from requests->tika) (3.2.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/alberli/opt/anaconda3/envs/pythoneleven/lib/python3.11/site-packages (from requests->tika) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /Users/alberli/opt/anaconda3/envs/pythoneleven/lib/python3.11/site-packages (from requests->tika) (2.0.4)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/alberli/opt/anaconda3/envs/pythoneleven/lib/python3.11/site-packages (from requests->tika) (2023.7.22)\n",
      "Installing collected packages: tika\n",
      "Successfully installed tika-2.6.0\n"
     ]
    }
   ],
   "source": [
    "! pip install tika"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "56fc46d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2\n"
     ]
    }
   ],
   "source": [
    "# Get cwd to place the tika app jar file\n",
    "print(os.getcwd())\n",
    "os.environ['TIKA_APP_JAR'] = '/Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI 550 HW2'\n",
    "# Import tika\n",
    "from tika import parser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7600270e",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting faker\n",
      "  Obtaining dependency information for faker from https://files.pythonhosted.org/packages/6c/1d/8030a4cccfe57b66125cf555ba71a2e5f7c96196cfe7d442b72e42da1230/Faker-19.2.0-py3-none-any.whl.metadata\n",
      "  Using cached Faker-19.2.0-py3-none-any.whl.metadata (15 kB)\n",
      "Requirement already satisfied: python-dateutil>=2.4 in /Users/alberli/opt/anaconda3/envs/pythoneleven/lib/python3.11/site-packages (from faker) (2.8.2)\n",
      "Requirement already satisfied: six>=1.5 in /Users/alberli/opt/anaconda3/envs/pythoneleven/lib/python3.11/site-packages (from python-dateutil>=2.4->faker) (1.16.0)\n",
      "Using cached Faker-19.2.0-py3-none-any.whl (1.7 MB)\n",
      "Installing collected packages: faker\n",
      "Successfully installed faker-19.2.0\n"
     ]
    }
   ],
   "source": [
    "# ! pip install faker"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59d52674",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# !python3 -m pip install PyPDF2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29be129c",
   "metadata": {},
   "source": [
    "## Solving fake author name issue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "70535c5a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "John Williams\n",
      "Wayne Velez\n",
      "Phyllis Henry\n",
      "Derek Wong\n",
      "Mason Hughes\n",
      "Shawn Jacobs\n",
      "Daniel Schultz\n",
      "Jennifer Ryan\n",
      "David Miller\n",
      "Jody Sharp\n"
     ]
    }
   ],
   "source": [
    "from faker import Faker\n",
    "fake = Faker()\n",
    "for _ in range(10):\n",
    "    print(fake.name())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "608af4bb",
   "metadata": {},
   "source": [
    "## Tika Parsing Part"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "034d01a8",
   "metadata": {},
   "source": [
    "### Using PyPDF2 to skip the first ResearchGate page inside the pdf files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "fab22c42",
   "metadata": {},
   "outputs": [],
   "source": [
    "import PyPDF2\n",
    "\n",
    "\n",
    "def extract_text_from_second_page(pdf_file_path, output_file_path):\n",
    "    # Open the PDF file\n",
    "    with open(pdf_file_path, 'rb') as pdf_file:\n",
    "        # Create a PDF reader object\n",
    "        pdf_reader = PyPDF2.PdfReader(pdf_file)\n",
    "        \n",
    "        # Check if the PDF has at least two pages\n",
    "        if len(pdf_reader.pages) >= 2:\n",
    "            # Initialize an empty string to store the text\n",
    "            extracted_text = \"\"\n",
    "            \n",
    "            # Start from the second page (index 1) and extract text from each page\n",
    "            for page in pdf_reader.pages[1:]:\n",
    "                extracted_text += page.extract_text()\n",
    "            \n",
    "            # Save the extracted text to the output file\n",
    "            with open(output_file_path, 'w', encoding='utf-8') as output_file:\n",
    "                output_file.write(extracted_text)\n",
    "                \n",
    "            print(f\"Text extracted from the second page and saved to {output_file_path}.\")\n",
    "        else:\n",
    "            print(\"PDF does not have at least two pages.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3224c934",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Text extracted from the second page and saved to /Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/trial.txt.\n"
     ]
    }
   ],
   "source": [
    "# Example usage\n",
    "pdf_file_path = '/Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/trial dataset/LRP-6_is_co-receptor_for_multiple_fibrogenic_pathw.pdf'\n",
    "\n",
    "# Replace 'output.txt' with the desired output file path\n",
    "output_file_path = '/Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/trial.txt'\n",
    "\n",
    "# Call the function to extract and save the text\n",
    "extract_text_from_second_page(pdf_file_path, output_file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "93c12d64",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "LRP-6 is a coreceptor for multiple ﬁbrogenic signaling\n",
      "pathways in pericytes and myo ﬁbroblasts that\n",
      "are inhibited by DKK-1\n",
      "Shuyu Rena,b, Bryce G. Johnsona,b, Yujiro Kidaa,b, Colin Ipa,b, Kathryn C. Davidsonc,d, Shuei-Liong Line, Akio Kobayashif,\n",
      "Richard A. Langg, Anna-Katerina Hadjantonakish, Randall T. Moonc,d, and Jeremy S. Duf ﬁelda,b,d,1\n",
      "aDivision of Nephrology andbCenter for Lung Biology, Department of Medicine and Pathology,cHoward Hughes Medical Institute, anddInstitute for Stem\n",
      "Cell and Regenerative Medicine, University of Washington, Seattle, WA 98109;eDepartment of Physiology and Medicine, National Taiwan University Hospital,\n",
      "Taipei 100, Taiwan;fRenal Division, Department of Medicine, Harvard Medical School, Boston, MA 02115;gCincinnati Children ’s Hospital Medical Center,\n",
      "Cincinnati, OH 45229; andhDevelopmental Biology Program, Sloan-Kettering Institute, New York, NY 10065\n",
      "Edited by Michael Karin, University of California, San Diego School of Medicine, La Jolla, CA, and approved December 7, 2012 (received for review\n",
      "June 29, 2012)\n",
      "Fibrosis of vital organs is a major public health problem with limited\n",
      "therapeutic options. Mesenchymal cells including microvascular\n",
      "mural cells (pericytes) are major progenitors of scar-forming my-\n",
      "oﬁbroblasts in kidney and other organs. Here we show pericytes in\n",
      "healthy kidneys have active WNT/ β-catenin signaling responses\n",
      "that are markedly up-regulated following kidney injury. Dickkopf-\n",
      "related protein 1 (DKK-1), a ligand for the WNT coreceptors low-\n",
      "density lipoprotein receptor-related proteins 5 and 6 (LRP-5 and LRP-\n",
      "6) and an inhibitor of WNT/ β-catenin signaling, effectively inhibits\n",
      "pericyte activation, detachment, and transition to myo ﬁbroblasts in\n",
      "vivo in response to kidney injury, resulting in attenuated ﬁbrogen-\n",
      "esis, capillary rarefaction, and in ﬂammation. DKK-1 blocks activa-\n",
      "tion and proliferation of established myo ﬁbroblasts in vitro and\n",
      "blocks pericyte proliferation to PDGF, pericyte migration, gene ac-\n",
      "tivation, and cytoskeletal reorganization to TGF- βor connective\n",
      "tissue growth factor. These effects are largely independent of\n",
      "inhibition of downstream β-catenin signaling. DKK-1 acts predom-\n",
      "inantly by inhibiting PDGF-, TGF- β–, and connective tissue growth\n",
      "factor-activated MAPK and JNK signaling cascades, acting via LRP-6\n",
      "with associated WNT ligand. Biochemically, LRP-6 interacts closely\n",
      "with PDGF receptor βand TGF- βreceptor 1 at the cell membrane,\n",
      "suggesting that it may have roles in pathways other than WNT/\n",
      "β-catenin. In summary, DKK-1 blocks many of the changes in\n",
      "pericytes required for myo ﬁbroblast transition and attenuates estab-\n",
      "lished myo ﬁbroblast proliferation/activation by mechanisms de-\n",
      "pendent on LRP-6 and WNT ligands but not the downstream\n",
      "β-catenin pathway.\n",
      "Fibrosis of the internal organs, resulting from subclinical injury\n",
      "to the organ over a period of time or from acute severe injury\n",
      "or inﬂammation, is a major global health problem. All organs may\n",
      "be affected by ﬁbrosis, which matures into microscopic or mac-\n",
      "roscopic scarring within the tissue parenchyma. At present there\n",
      "are few therapies that speci ﬁcally target the process of ﬁbro-\n",
      "genesis, despite increasing evidence suggesting that ﬁbrogenesis\n",
      "per se provokes further decline in organ function, in ﬂammation,\n",
      "and tissue ischemia (1). In addition, myo ﬁbroblasts themselves\n",
      "are in ﬂammatory cells that generate cytokines, chemokines, and\n",
      "radicals that promote injury. Myo ﬁbroblasts appear as a result of\n",
      "a transition from pericytes, cells that normally nurse, maintain,\n",
      "and regulate the microvasculature (2, 3). The transition from\n",
      "pericytes to myo ﬁbroblasts results in an unstable microvascula-\n",
      "ture leading to aberrant angiogenesis or rarefaction (3). These\n",
      "microvascular changes ultimately provoke tissue ischemia. There-\n",
      "fore, the myo ﬁbroblast and its transition from resident pericyte or\n",
      "ﬁbroblast is a major new target for therapeutics to counter the\n",
      "deleterious consequences of tissue injury.\n",
      "Recently, SNPs in LRP-6, a transmembrane coreceptor for\n",
      "WNTs that binds to Frizzled (Frz) receptors and thereby to\n",
      "the WNT/ β-catenin signaling cascade, have been identi ﬁed as\n",
      "independent risk factors for cardiovascular diseases. However,\n",
      "the mechanisms still are obscure (4, 5).The WNT/ β-catenin signaling pathway is a major regulator of\n",
      "cell function both in embryonic development and in adults. Both\n",
      "elevated and attenuated levels of signaling that fall outside the\n",
      "normal homeostatic range of WNT signaling are linked to ab-\n",
      "normal embryonic development and to diverse disease states (6).\n",
      "Increasing evidence indicates that WNT signaling plays critical\n",
      "roles in tissue regeneration and immune responses to injury and\n",
      "infection (7). However, the signaling cascade and the cellular\n",
      "responses are complex and context speci ﬁc( 8 ) .\n",
      "Previous studies have highlighted the importance of WNT/\n",
      "β-catenin in kidney regeneration (9), and studies of chronic dis-\n",
      "ease of the kidney glomerulus and liver sinusoids suggest that\n",
      "persistent activation of WNT/ β-catenin is deleterious (10), but the\n",
      "role of WNT signaling in myo ﬁbroblasts and their precursors,\n",
      "pericytes of the kidney capillaries, was not studied (2). Here we\n",
      "explore the consequences of reactivating WNT/ β-catenin signal-\n",
      "ing in pericytes and myo ﬁbroblasts after kidney injury.\n",
      "Results and Discussion\n",
      "Wnt/β-Catenin Pathway Is Up-Regulated in Myo ﬁbroblasts in Kidney\n",
      "Disease. To explore the extent of the activation of the canonical\n",
      "WNT pathway in cells of the normal kidney, we studied two\n",
      "distinct lines of mice that are transgenic for reporters of WNT/\n",
      "β-catenin signaling. Axin2+/laczgenerates β-gal in cells expressing\n",
      "the endogenous WNT/ β-catenin target gene, Axin2. (Fig. 1 A)\n",
      "(11). TCF/LEF-H2B-GFPTris a recently validated transgenic line\n",
      "of mice reporting β-catenin activity by nuclear GFP expression. It\n",
      "exhibits enhanced sensitivity and speci ﬁcity over previous re-\n",
      "porters of β-catenin nuclear activity (Fig. 1 A) (12). In normal\n",
      "kidneys of Axin2+/laczmice there is extensive WNT/ β-catenin\n",
      "signaling in the papilla, and there are more restricted responses\n",
      "in the normal medulla and kidney cortex, as we have demon-\n",
      "strated previously ( Fig. S1 A) (9). Among the cells expressing\n",
      "lacZ were podocytes, vascular smooth muscle of arterioles, and\n",
      "some pericytes (Fig. 1 B–DandFig. S1 AandB). In response to\n",
      "injury initiated by obstructing urine ﬂow from the kidney (the\n",
      "unilateral ureteric obstruction [UUO] model), there was marked\n",
      "increase in lacZ staining in the scar-forming cells known as\n",
      "“myoﬁbroblasts ”that derive from pericytes (Fig. 1 B–D). Although\n",
      "collecting duct cells and cells of the loop of Henle showed Axin2 -\n",
      "Author contributions: S.R. and J.S.D. designed research; S.R., B.G.J., C.I., S.-L.L., and J.S.D.\n",
      "performed research; Y.K., K.C.D., S.-L.L., A.K., R.A.L., A.-K.H., R.T.M., and J.S.D. contrib-\n",
      "uted new reagents/analytic tools; S.R., B.G.J., C.I., and J.S.D. analyzed data; and K.C.D.,\n",
      "R.T.M., and J.S.D. wrote the paper.\n",
      "Conﬂict of interest statement: S.R. and J.S.D have submitted patent applications for the\n",
      "use of DKK-1 in ﬁbrosis.\n",
      "This article is a PNAS Direct Submission.\n",
      "1To whom correspondence should be addressed. E-mail: jeremysd@uw.edu.\n",
      "This article contains supporting information online at www.pnas.org/lookup/suppl/doi:10.\n",
      "1073/pnas.1211179110/-/DCSupplemental .\n",
      "1440 –1445 |PNAS |January 22, 2013 |vol. 110 |no. 4 www.pnas.org/cgi/doi/10.1073/pnas.1211179110lacZ staining, little lacZ was detected in distal and proximal\n",
      "tubules in either normal or diseased kidneys ( Fig. S1 B).\n",
      "To visualize this WNT response in greater detail, we localized\n",
      "WNT reporter activity in TCF/LEF-H2B-GFPTrmice. In normal\n",
      "kidney, β-catenin responses were more extensive than previously\n",
      "appreciated. We have shown previously that in normal kidney,\n",
      "proximal and distal epithelium exhibits little endogenous WNT/\n",
      "β-catenin reporter activity (9). In contrast, in TCF/LEF-H2B-GFPTr\n",
      "mice there was evidence of WNT re porter activity in many of these\n",
      "cells (Fig. 1 EandFig. S1 C–G). However, like the Axin2+/laczre-\n",
      "porter, a minority of pericytes showed active signaling in normal\n",
      "kidney. Following kidney injury, WNT/ β-catenin reporter activity\n",
      "increased in epithelial cell compartments, but there was a much\n",
      "greater increase in the pericyte/myo ﬁbroblast population of cells\n",
      "(Fig. 1 E–GandFig. S1 C).\n",
      "Kidney Myo ﬁbroblasts Up-Regulate Wnt Pathway Genes and Wnt\n",
      "Reporter Activity. To explore further the activation of the WNT\n",
      "pathway in kidney myo ﬁbroblasts, we puri ﬁed nonactivated peri-\n",
      "cytes (normal kidney), activated pericytes (from day 2 after\n",
      "UUO), and myo ﬁbroblasts transitioned from pericytes (from days\n",
      "5, 7, and 10 after UUO) by ﬂow cytometric sorting of single-cell\n",
      "digests from kidneys of the Coll-GFPTrreporter mouse, which\n",
      "speciﬁcally expresses GFP in these cells (13). Puri ﬁed cells showed\n",
      "marked up-regulation of pericyte genes, including P75 NGFR\n",
      "(Ngfr)(Fig. S2 A), and at baseline expressed detectable levels of\n",
      "NG2 ( Cspg4 ) and PDGFR β(Pdgfrb ) that were modestly up-reg-\n",
      "ulated over time postinjury ( Fig. S2 D,Table S1 ). We found pro-\n",
      "totypical pro ﬁbrotic genes expressed in normal kidney and up-\n",
      "regulated over time postinjury in our disease model, including\n",
      "Col1a1 ,Tgfb1 ,Ctgf,a n d Fgf2. We also observed up-regulation of\n",
      "the phosphatonin Fgf23 , which has been implicated in the path-\n",
      "ogenesis of cardiovascular disease ( Fig. S2 A). WNT ligands, in-\n",
      "cluding Wnt2 ,3,7b,8a,8b, and 10a, were up-regulated. Pericytes\n",
      "also express cell-surface receptors (Fzd) and coreceptors (Lrp5,Lrp6) for WNT responsiveness ( Fig. S2 B). As the course of kidney\n",
      "disease progressed, there was a modest up-regulation of the\n",
      "Dickkopf-related protein (DKK) family of mRNAs known to in-\n",
      "hibit canonical WNT signaling via direct binding to the LRP-5 and\n",
      "LRP-6 coreceptors ( Fig. S2 C). Other WNT downstream target\n",
      "genes, including sFLT1 and WISP1, which play important roles in\n",
      "noncanonical and canonical WNT signaling pathways, re-\n",
      "spectively, also were up-regulated, consistent with activated WNT/\n",
      "β-catenin and also noncanonical WNT pathways ( Fig. S2 C).\n",
      "DKK-1 Triggers G1 Cell-Cycle Arrest and Down-Regulates Activation in\n",
      "Myoﬁbroblasts. To test the function of WNT/ β-catenin pathway\n",
      "activation in myo ﬁbroblasts, we generated primary myo ﬁbroblast\n",
      "cultures from Coll-GFPTrmice that had kidney ﬁbrosis (13).\n",
      "Myoﬁbroblasts also expressed WNT ligands and receptors ( Fig.\n",
      "S3A). DKK-1 has been reported to inhibit the WNT/ β-catenin\n",
      "pathway by binding to LRP-5 or LRP-6. We cloned Dkk1 and\n",
      "expressed soluble DKK-1 or the DKK-1-GFP fusion protein\n",
      "(Fig. S3 B) in HEK293 cells. Soluble DKK-1-GFP binding to\n",
      "unlabeled cell lines was markedly enhanced by overexpression of\n",
      "LRP-5 or LRP-6 ( Fig. S3 C), and DKK-1-GFP readily bound to\n",
      "the surface of cultured myo ﬁbroblasts without the requirement\n",
      "for receptor overexpression ( Fig. S3 D), indicating high receptor\n",
      "density for DKK-1 on myo ﬁbroblasts. DKK-1 protein speci ﬁcally\n",
      "blocked entry of myo ﬁbroblasts into the cell cycle induced by\n",
      "serum at the G1 checkpoint (Fig. 2 A–FandFig. S3 E), resulting\n",
      "in inability to proliferate (Fig. 2 B).\n",
      "To understand the mechanisms better, we tested the effect of\n",
      "DKK-1 on the steady-state level of β-catenin protein. Cytosolic\n",
      "and nuclear β-catenin increased 1 h and 16 h after serum acti-\n",
      "vation, an effect decreased by DKK-1 at 16 h (Fig. 2 G), con-\n",
      "ﬁrming DKK-1 inhibits WNT/ β-catenin signaling but suggesting\n",
      "early DKK-1 responses may occur independent of regulation of\n",
      "β-catenin protein (Fig. 2 G). Transcripts for αSMA ( Acta2 ), a\n",
      "marker of myo ﬁbroblast activation, were increased by serum at\n",
      "24 h, and DKK-1 signi ﬁcantly reduced this activation (Fig. 2 H),\n",
      "suggesting that DKK-1 may regulate activation also.\n",
      "Systemic Delivery of DKK-1 Inhibits Myo ﬁbroblast Expansion and\n",
      "Fibrosis. Next we tested whether DKK-1 inhibited pericytes and\n",
      "myoﬁbroblasts in vivo. DKK-1 was delivered systemically using an\n",
      "Fig. 1. WNT/ β-catenin signaling is activated during kidney injury in the\n",
      "pericyte/myo ﬁbroblast cell compartment. ( A) Schema showing the Axin2LacZ\n",
      "allele and the TCF/Lef:H2B-GFP transgene, which report WNT/ β-catenin sig-\n",
      "naling. ( B–D) WNT responses identi ﬁed by blue stain or green ﬂuorescence in\n",
      "cells ( B) and quanti ﬁed in graphs ( Cand D) during kidney injury induced by\n",
      "UUO in Axin2+/laczreporter mice are seen predominantly in myo ﬁbroblasts\n",
      "(arrowheads), but no stain is seen in Axin2+/+kidneys. ( E–G) WNT/ β-catenin\n",
      "responses identi ﬁed by nuclear GFP in confocal images ( E) of kidneys from\n",
      "TCF/Lef:H2B-GFPTrreporter mice after UUO, highlighting PDGFR β+cells or\n",
      "αSMA cells, two markers for myo ﬁbroblasts in diseased kidney, and quan-\n",
      "tiﬁed (Fand G)b yP D G F R β-o rαSMA-staining cells expressing nuclear GFP.\n",
      "*P<0.05, ** P<0.01. n=4 per group. Error bars indicate SEM.\n",
      "Fig. 2. DKK-1 binds to myo ﬁbroblasts and blocks proliferation by G1/S cell-\n",
      "cycle arrest in vitro. ( A) Graph showing BrdU nuclear incorporation in qui-\n",
      "escent myo ﬁbroblasts stimulated for 3 h with 3% FCS and DKK-1 or vehicle.\n",
      "(B) Coulter-counted kidney quiescent myo ﬁbroblasts stimulated for 24 h\n",
      "with 3% FCS and DKK-1 or vehicle. ( Cand D) Flow cytometric plots ( C)a n d\n",
      "graph ( D) showing BrdU uptake in myo ﬁbroblasts stimulated for 3 h with\n",
      "3% FCS and DKK-1 or vehicle. ( EandF) Propidium iodide DNA content plots\n",
      "(E) and graph ( F) showing quiescent myo ﬁbroblasts stimulated for 24 h with\n",
      "3% FCS and DKK-1 or vehicle. ( G) The effect of DKK-1 on cytoplasmic and\n",
      "nuclear β-catenin protein. Serum increases β-catenin, an effect not modu-\n",
      "lated by DKK-1 at 1 h, but DKK-1 markedly reduces β-catenin levels at later\n",
      "time points. ( H) qPCR data showing myo ﬁbroblast expression of Acta2 after\n",
      "treatment with FCS or FCS +DKK-1. * P<0.05, ** P<0.01, *** P<0.001. n=4\n",
      "per group. Error bars indicate SEM.\n",
      "Ren et al. PNAS |January 22, 2013 |vol. 110 |no. 4 |1441\n",
      "MEDICAL SCIENCESadenoviral delivery system that generates high-level expression of\n",
      "circulating DKK-1 protein (Fig. 3 A). We tested circulating DKK-1\n",
      "on both the development and progression of kidney ﬁbrosis using\n",
      "three different experimental designs (Fig. 3 B). In preventative\n",
      "studies using the UUO model, DKK-1 profoundly inhibited peri-\n",
      "cyte expansion, proliferation, and transition to myo ﬁbroblasts (Fig.\n",
      "3C–I), resulting in reduced ﬁbrosis ( Fig. S4 ). The inhibition on\n",
      "day 4 after UUO was associated with a marked reduction in\n",
      "inﬂammation (Fig. 3 J) and epithelial injury ( Fig. S4 AandB).\n",
      "Moreover, expression of the WNT/ β-catenin reporters in kidney\n",
      "myoﬁbroblasts was reduced, particularly at the earlier time points\n",
      "(Fig 3 K). These studies were extended to day 10 after UUO in-\n",
      "jury, and there was a substantial reduction in organ ﬁbrosis and\n",
      "collagen transcripts (Fig. 3 L–M).\n",
      "To study this effect in more detail, we performed a reversal\n",
      "study in which DKK-1 was delivered after disease onset, and ﬁ-\n",
      "brosis extent was assessed 10 d after UUO (Fig. 3 N–P). Myo ﬁ-\n",
      "broblast accumulation, proliferation, deposition of interstitial\n",
      "ﬁbrosis, and rarefaction of the capillaries were inhibited. Fur-\n",
      "thermore, as in the preventative studies, DKK-1 inhibited in-\n",
      "ﬂammation and epithelial injury ( Fig. S4 A–C). Because recent\n",
      "evidence indicates that myo ﬁbroblasts in kidney arise from peri-\n",
      "cyte precursors which detach from peritubular capillaries in re-\n",
      "sponse to injury (2), we examined the effect of DKK-1 on pericyte\n",
      "precursor detachment, spreading, and migration in Coll-GFPTr\n",
      "reporter mice. DKK-1 inhibited pericyte detachment from capil-\n",
      "laries at early time points in this model (Fig. 3 Q).\n",
      "We tested whether DKK-1 could inhibit disease in a second\n",
      "model of in ﬂammation and ﬁbrosis, ischemia reperfusion injury\n",
      "(U-IRI) of a single (unilateral) kidney, a recognized model of\n",
      "chronic kidney disease following acute kidney injury (9). We\n",
      "performed a reversal study in this model and characterized the\n",
      "extent of scarring in the kidney 10 d after the initial U-IRI. Cir-\n",
      "culating DKK-1 inhibited myo ﬁbroblast accumulation, in ﬂamma-\n",
      "tion, and interstitial ﬁbrosis and reduced epithelial injury ( Fig. S4\n",
      "E–K) sufﬁciently to reverse existent ﬁbrosis ( Fig. S4 L).\n",
      "DKK-1 Blocks PDGF-Stimulated Proliferation of Kidney Pericytes by an\n",
      "LRP-6, P42/P44 MAPK-Dependent Mechanism. To study the mecha-\n",
      "nism of action of DKK-1, we tested its effects on primary kidney\n",
      "pericyte cultures, the precursors of myo ﬁbroblasts (2, 13). PDGF\n",
      "and TGF- βsignaling in pericytes are important factors in the\n",
      "detachment of pericytes from capillaries and their transition to\n",
      "myoﬁbroblasts (3, 14) and as such are important factors in\n",
      "driving ﬁbrosis. Connective tissue growth factor (CTGF) and its\n",
      "homolog CCN1 (Cyr61) also have been implicated in ﬁbrosis and\n",
      "wound healing (15). Pericytes were stimulated by PDGF-BB to\n",
      "enter the cell cycle (Fig. 4 A), but other pro ﬁbrogenic growth fac-\n",
      "tors had no effect. Unexpectedly, exogenous addition of WNT3a\n",
      "did not stimulate proliferation. DKK-1 markedly attenuated\n",
      "PDGF-BB –stimulated proliferation (Fig. 4 B). PDGF stimulation\n",
      "down-regulated transcripts for the activation markers CTGF,\n",
      "αSMA, and collagen I α(1), and DKK-1 further down-regulated\n",
      "those transcripts (Fig. 4 C), suggesting that DKK-1 therefore may\n",
      "enhance certain PDGF-mediated transcriptional events. None of\n",
      "the factors applied to pericytes affected viability (Fig. 4 D).\n",
      "In keeping with a link between PDGF signaling and the WNT\n",
      "pathway in pericyte proliferation, PDGF regulates a number of\n",
      "WNT ligands, including up-regulating Wnt2 and 5aand down-\n",
      "regulating Wnt7a ,10a, and 16(Fig. 4 E). However, inconsistent\n",
      "with such a link, exogenous WNT3a had no stimulatory effect on\n",
      "proliferation (Fig. 4 A). Because DKK-1 acts via LRP-5 and\n",
      "LRP-6, and because PDGF regulates WNT transcripts (Fig. 4 E),\n",
      "we expected DKK-1 to block the endogenous WNT/ β-catenin\n",
      "Fig. 3. DKK-1 blocks pericyte activation and transition to myo ﬁbroblasts\n",
      "and reverses myo ﬁbroblast activation in vivo, inhibiting ﬁbrogenesis, capil-\n",
      "lary rarefaction, and in ﬂammation. ( A) Western blots of 5 μL of plasma from\n",
      "mice 5 d after i.v. injection of Ad-control or Ad-DKK-1 and from mice sub-\n",
      "jected to sham surgery and injected with control. ( B) Experimental schemata\n",
      "for adenoviral administration, kidney injury, and analysis in the UUO model.\n",
      "(C–M) Prevention studies. ( C) Low-magni ﬁcation confocal images of kidney\n",
      "cortex 4 d after sham operation or UUO in Coll-GFPTrmice that had received\n",
      "Ad-control or Ad-DKK-1 6 d previously, showing Coll-GFP cells or PDGFR β\n",
      "cells. g, glomerulus; a, arteriole. ( D–F) Graphs showing quanti ﬁcation of Coll-\n",
      "GFP cells, PDGFR βcells, and αSMA cells in kidney 4 d after UUO. ( G) Pro-\n",
      "portion of Coll-GFP cells that express the proliferation marker Ki67. ( HandI)\n",
      "Western blot of GFP ( H)o rαSMA/CTGF ( I) in whole Coll-GFP mouse kidney\n",
      "4 d after UUO. ( J) Quanti ﬁcation of macrophage numbers in kidney sections\n",
      "detected by F4/80 staining. ( K) Western blot quantifying canonical WNT\n",
      "signaling by detecting the H2B-GFP fusion protein after Ad-DKK-1 vs. Ad-\n",
      "control treatment of TCF/Lef:H2B-GFPTrreporter mice during UUO kidney\n",
      "injury. ( L) Sirius red-stained kidneys 10 d after UUO. ( M) Morphometry of\n",
      "Sirius red-stained collagen ( Upper ) or qPCR for Col1a1 transcripts ( Lower )\n",
      "10 d after UUO in mice treated with Ad-control vs. Ad-DKK-1. ( N–P) Reversal\n",
      "studies. Confocal Images ( N) and morphometric quanti ﬁcation ( O)o fαSMA\n",
      "staining 10 d after UUO in mice treated with Ad-control or Ad-DKK-1 from\n",
      "day 4. ( P) Quanti ﬁcation of capillary density 10 d after UUO. Note thatrarefaction occurs in response to kidney disease, but DKK-1 partially reverses\n",
      "rarefaction. ( Q) Pericyte detachment. Images and quanti ﬁcation of pericyte\n",
      "area in Coll-GFP mice 2 d after UUO in the presence of circulating DKK-1 or\n",
      "control. Note that injury to the kidney stimulated pericyte spreading and\n",
      "detachment from endothelium (arrowheads). * P<0.05, ** P<0.01. n=4–6\n",
      "per group. Error bars indicate SEM.\n",
      "1442 |www.pnas.org/cgi/doi/10.1073/pnas.1211179110 Ren et al.pathway. Binding of WNT ligands to Fzd receptors and LRP-6\n",
      "coreceptors leads to activation and phosphorylation of LRP-6\n",
      "(pLRP-6). Surprisingly therefore, PDGF-BB alone stimulatespLRP-6 (Fig. 4 F), and DKK-1 initially augments pLRP-6 mod-\n",
      "estly and inhibits it only at later time points (Fig. 4 FandFig. S5\n",
      "A–C), indicating that PDGF may coactivate the WNT/ β-catenin\n",
      "pathway and that DKK-1 regulates this coactivation. To test this\n",
      "possibility, we assessed whether PDGF-BB regulates the WNT/\n",
      "β-catenin pathway in pericytes cultured from the β-catenin re-\n",
      "porter mouse ( TCF/Lef:H2B-GFPTr)( F i g .4 G). After 16 h with\n",
      "PDGF, nuclear β-catenin activity was increased signi ﬁcantly.\n",
      "This increase was inhibited by DKK-1 (Fig. 4 G), although the\n",
      "extent of β-catenin activation was much lower than the extent of\n",
      "cells triggered into the cell cycle (Fig. 4 A), aﬁnding that is\n",
      "consistent with the possibility that DKK-1 inhibits proliferation\n",
      "by a β-catenin –independent pathway.\n",
      "We therefore explored signaling pathways activated by PDGF-\n",
      "BB but inhibited by DKK-1 (Fig. 4 HandFig. S5 D). PDGF-BB\n",
      "stimulates the P42/P44 MAPK and the JNK signaling pathways.\n",
      "DKK-1 inhibits both these responses. PDGF-BB activates the P38\n",
      "MAPK pathway, but DKK-1 enhances this response, perhaps\n",
      "explaining why DKK-1 can enhance PDGF-mediated transcrip-\n",
      "tion (Fig. 4 C). As expected, PDGF-BB enhances cyclin D1 ex-\n",
      "pression (Fig. 4 HandFig. S5 D), an effect inhibited by DKK-1.\n",
      "Although WNT3a alone had no apparent impact on cell prolif-\n",
      "eration (Fig. 4 A), it nevertheless stimulated the activation of\n",
      "LRP-6 and accumulation of cyclin D1 ( Fig. S5 E), providing ev-\n",
      "idence that LRP-6 activation by PDGF-BB results in distinct\n",
      "signaling vs. activation by WNT3a.\n",
      "Because DKK-1 speci ﬁcally inhibits PDGF-stimulated P42/\n",
      "P44 MAPK and JNK activation (Fig. 4 H), the relative impor-\n",
      "tance of these pathways was tested using speci ﬁc inhibitors. The\n",
      "P42/P44 inhibitor U0126 completely replaced DKK-1 function in\n",
      "PDGF-BB –stimulated pericyte proliferation (Fig. 4 I), whereas\n",
      "the JNK inhibitor SP600125 had no impact. Because there was\n",
      "no apparent linkage between the PDGF-BB –mediated activation\n",
      "of the WNT/ β-catenin signaling pathway and its capacity to\n",
      "stimulate proliferation, we tested whether the WNT/ β-catenin\n",
      "pathway is necessary for PDGF-mediated proliferation using\n",
      "a small-molecule inhibitor of β-catenin, XAV939 (a Tankyrase\n",
      "inhibitor). Strikingly, XAV939 had no effect (Fig. 4 I). To con-\n",
      "ﬁrm this ﬁnding, cultured pericytes from kidneys homozygous for\n",
      "theﬂoxed alleles of β-catenin ( Ctnnb1ﬂ/ﬂ) underwent recombi-\n",
      "nation in vitro by transduction with Lenti-Cre virus, and Lenti-\n",
      "GFP virus was used as a control that does not catalyze recom-\n",
      "bination. One hundred percent of pericytes were transduced by\n",
      "GFP expression, and β-catenin protein was lost 48 h after trans-\n",
      "duction with Lenti-Cre ( Fig. S6 ). Pericytes lacking β-catenin\n",
      "responded to PDGF and its inhibition by DKK-1 similarly to\n",
      "pericytes with β-catenin (Fig. 4 J), con ﬁrming that PDGF stim-\n",
      "ulates and DKK-1 inhibits proliferation independently of the\n",
      "canonical WNT pathway, even though LRP-6 is activated. To\n",
      "study whether DKK-1 mediates its effects through an alternative\n",
      "cell-surface signaling mechanism or whether LRP-6 is necessary\n",
      "for the PDGF signaling pathway in this context, we immunopre-\n",
      "cipitated PDGFR βfrom pericytes and pulled down active pLRP-\n",
      "6 only when the receptor was engaged with ligand (Fig. 4 K), in-\n",
      "dicating a close relationship between the two receptors at the time\n",
      "of signaling. Next, we overexpressed dominant-negative mutated\n",
      "forms of LRP-6 in a mouse embryonic ﬁbroblast cell line (3T3)\n",
      "that endogenously expresses PDGFR β(Fig. 4 Land Fig. S6 ).\n",
      "Expression of two different dominant-negative forms of LRP-6,\n",
      "but not the WT form, was suf ﬁcient to inhibit proliferation of 3T3\n",
      "cells in response to PDGF-BB and to prevent DKK-1 effects (Fig.\n",
      "4L), indicating that DKK-1 inhibits proliferation through signal-\n",
      "ing via LRP-6 and revealing that LRP-6 can be detected in a\n",
      "complex containing PDGFR β. Finally, because LRP-6 is required\n",
      "for PDGF-induced proliferation, we assessed whether WNT\n",
      "ligands are necessary for PDGF responses by blocking WNT se-\n",
      "cretion in pericytes using the Porcupine homolog inhibitor, IWP2\n",
      "(Fig. S7 ). Porcupine homolog is required for palmitoylation and\n",
      "secretion of all WNTs from cells (16). IWP2, blocked PDGF-\n",
      "stimulated proliferation in a dose-dependent manner without\n",
      "affecting viability, suggesting that WNT engagement of LRP-6 at\n",
      "the cell surface is necessary for PDGF responses in pericytes.\n",
      "Fig. 4. DKK-1 inhibits PDGF-BB –mediated proliferation of pericytes in vitro\n",
      "by a noncanonical, LRP-6 –dependent, P42/44 MAPK-dependent mechanism.\n",
      "(A) Graph of BrdU incorporation into quiescent kidney pericytes 6 h after\n",
      "stimulation with cytokines. ( B) The effect of DKK-1 on PDGF-BB –stimulated\n",
      "proliferation. ( C) qPCR of genes associated with cell activation in pericytes\n",
      "48 h after stimulation. ( D)Q u a n t i ﬁcation of cell viability in pericytes stimulated\n",
      "with cytokines and DKK-1 for 24 h. ( E) RT-PCR results showing the effect of\n",
      "PDGF-BB on WNT ligands and receptors in pericytes 12 h after stimulation. C,\n",
      "control; P, PDGF; arrowheads indicate regulated genes. ( F) Western blot\n",
      "time course showing pPDGFR βand pLRP-6 levels in pericytes. DKK-1 does not\n",
      "affect pLRP-6 at early time points but inhibits pLRP-6 at later time points. ( G)\n",
      "Fluorescence images and data quantifying nuclear GFP+(green) in TCF/Lef:\n",
      "H2B-GFPTrcanonical WNT reporter pericytes 16 h after PDGF-BB or PDGF-BB +\n",
      "DKK-1. ( H) Western blot time course of phosphorylated forms of P42/P44, JNK,\n",
      "and P38, PDGFR β, and total cyclinD1 in pericytes activated by PDGF-BB or\n",
      "PDGF-BB +DKK-1. ( I) Graph showing the effect of DKK-1 or the canonical WNT\n",
      "inhibitor XAV939, the P42/P44 inhibitor U0126, or the JNK inhibitor SP600125\n",
      "on PDGF-BB –stimulated BrdU incorporation into quiescent pericytes. ( J)Graph\n",
      "showing the effect of PDGF-BB on the proliferation of Ctnnb1ﬂ/ﬂpericytes that\n",
      "underwent in vitro recombination by expressing Cre recombinase vs. Ctnnb1ﬂ/ﬂ\n",
      "pericytes that expressed control protein GFP. ( K) Western blot of pericyte pro-\n",
      "teins immunoprecipitated by anti-PDGFR βantibodies or control antibodies\n",
      "detecting pLRP-6 or PDGFR β.(L) Graph showing the effect of expression of LRP-6\n",
      "(wild type) or dominant-negative forms of LRP-6, LRP-6 with tyrosine-to-me-\n",
      "thionine mutations at the ﬁve tyrosine sites (5m), or LRP-6 lacking the cyto-\n",
      "plasmic tail ( ΔC) on 3T3 ﬁbroblast proliferation in response to PDGF-BB and\n",
      "DKK-1. * P<0.05, ** P<0.01, *** P<0.01. n=4–7 per group. All blots are\n",
      "representative of three experiments. (Scale bars, 25 μm.) Error bars indicate SEM.\n",
      "Ren et al. PNAS |January 22, 2013 |vol. 110 |no. 4 |1443\n",
      "MEDICAL SCIENCESDKK-1 Inhibits TGF- β–and CTGF-Stimulated Activation of Kidney\n",
      "Pericytes via an LRP-6, JNK-Dependent Mechanism. O u ri nv i v os t u d -\n",
      "ies showed that DKK-1 prevents pericyte activation, detachment\n",
      "from capillaries, migration, and expression of the myo ﬁbroblast\n",
      "marker αSMA (Fig. 3). Previous and current studies (2, 14)\n",
      "suggest that PDGFR signaling does not affect these pericyte\n",
      "changes directly. Therefore we tested the effect of DKK-1 on\n",
      "pericyte activation and migration in response to other cytokines\n",
      "implicated in these processes (17, 18). In contrast to PDGF,\n",
      "TGF- βmarkedly and rapidly stimulates pericyte migration (Fig. 5\n",
      "A–C), and over 48 –72 h up-regulates collagen genes and the in-\n",
      "termediate ﬁlament αSMA (Fig. 5 D). Similar observations on\n",
      "migration were made by treatment of pericytes with CTGF (Fig. 5\n",
      "CandD), an extracellular protein that may signal via β1-integrins,\n",
      "LRP-1, and possibly LRP-6 (19), and with CCN1 ( Fig. S8 A).\n",
      "TGF- βand CTGF promote marked cytoskeletal reorganization\n",
      "of contractile ﬁlaments in pericytes after 24 h of cytokine treat-\n",
      "ment (Fig. 5 E). DKK-1 inhibits all migratory, activatory, and\n",
      "cytoskeletal changes in pericytes in response to TGF- βor CTGF\n",
      "(Fig. 5 A–E) but has no impact on migration in response to CCN1\n",
      "(Fig. S8 A). Strikingly, primary cultures of kidney epithelial cells\n",
      "have high endogenous migration that is weakly responsive to\n",
      "these cytokines. DKK-1 has a nonsigni ﬁcant effect on these cul-\n",
      "tures ( Fig. S3 ), suggesting DKK-1 has a major effect on pericytes,\n",
      "not epithelium.\n",
      "We next tested the impact of DKK-1 on activation of LRP-6\n",
      "in the context of these activating ligands. TGF- β, and CTGF both\n",
      "activate LRP-6 within minutes of cytokine exposure, and, as ex-\n",
      "pected, WNT3a also stimulates LRP-6 activation (Fig. 5 Fand\n",
      "Figs. S5 EandS8B). Within 10 min, DKK-1 weakly activates LRP-\n",
      "6(Fig. S5 C) and does not block the activation triggered by TGF- β\n",
      "or WNT3a (Fig. 5 FandFig. S8 ). However, LRP-6 was deactivated\n",
      "by DKK-1 in all activation conditions after 16 h ( Fig. S5 ), indi-\n",
      "cating LRP-6 is activated rapidly by non-WNT ligands and DKK-1\n",
      "either augments that signal weakly or does not affect it initially but\n",
      "silences it only at later time points. Because DKK-1 blocks TGF-\n",
      "β–mediated changes to pericytes, we tested whether TGF- βsignals\n",
      "via the WNT/ β-catenin pathway directly or indirectly. TGF- βreg-\n",
      "ulates WNT ligands in pericytes at 24 h (up-regulating Wnt2 ,9a,\n",
      "and11and down-regulating Wnt2b and16)( F i g .5 G), but DKK-1\n",
      "had no impact on these changes. Both TGF- βand CTGF triggered\n",
      "nuclear β-catenin activity in pericytes (Fig. 5 H); this activity was\n",
      "inhibited by DKK-1, but, as with PDGF, the effect occurred in\n",
      "a minority of cells, suggesting that TGF- βand CTGF may regulate\n",
      "pericyte activation via a WNT/ β-catenin –independent pathway and\n",
      "that DKK-1 may inhibit pericyte activation via that same pathway.\n",
      "We tested the effect of DKK-1 on TGF- β–and CTGF-medi-\n",
      "ated signaling. Both TGF- βand CTGF stimulate the P42/P44,\n",
      "P38, and JNK pathways. DKK-1 blocks activation of the P42/44\n",
      "and JNK pathways and augments activation of the P38 pathway\n",
      "(Fig. 5 IandJ). As expected, FAK is activated by both TGF- β\n",
      "and CTGF and is inhibited by DKK-1 (Fig. 5 IandJ). To test\n",
      "whether components of these DKK-1 –regulated signaling cas-\n",
      "cades are responsible for the impact on migration, we measured\n",
      "the effect of DKK-1 on TGF- βactivation of the canonical SMAD\n",
      "activation pathway ( Fig. S8 C), which was not affected. Next\n",
      "we used small-molecule inhibitors of activation of TGF βR1\n",
      "(SB431542), JNK (SP600125), P42/P44 (U0126), or β-catenin\n",
      "(XAV939) in TGF- β–or CTGF-stimulated migration assays.\n",
      "P42/P44, TGF βR1, and β-catenin inhibitors had no signi ﬁcant\n",
      "effect on TGF- β–stimulated migration, but JNK inhibitors\n",
      "blocked migration to TGF- βcompletely (Fig. 5 KandFig. S8 D),\n",
      "suggesting DKK-1 –mediated silencing of JNK signaling is central\n",
      "to its ability to block TGF- β–stimulated migration. Inhibition of\n",
      "CTGF-mediated migration was independent of TGF βR1 activa-\n",
      "tion, was partially dependent on β-catenin and P42/P44 activa-\n",
      "tion, and was most dependent on JNK activation (Fig. 5 L),\n",
      "highlighting similarities to and differences from TGF- β–mediated\n",
      "migration. CCN1-mediated migration occurred independently of\n",
      "P42/44 and JNK pathways ( Fig. S8 A). Because XAV939 showed\n",
      "little effect on TGF- β–mediated migration, we generated peri-\n",
      "cytes lacking β-catenin as described above (Fig. 4 J). Comparedwith controls, in the absence of β-catenin, DKK-1 completely\n",
      "blocked TGF- β–stimulated migration (Fig. 5 M). However,\n",
      "overall (nonstimulated) migration was reduced in the absence of\n",
      "β-catenin, indicating that DKK-1 can function independently of\n",
      "WNT/ β-catenin signaling but that the Wnt/ β-catenin pathway\n",
      "Fig. 5. DKK-1 blocks TGF- β–and CTGF-mediated migration of pericytes in\n",
      "vitro predominantly by a noncanonical, LRP-6 –dependent, JNK-dependent\n",
      "mechanism. ( Aand B) Images ( A) and time course graph ( B) showing mi-\n",
      "gration of kidney pericytes induced by TGF- βand blocked by DKK-1. (Scale\n",
      "bars, 50 μm.) (C) Graph of migration at 16 h by pericytes stimulated by TGF- β,\n",
      "CTGF, or WNT3a. All are blocked by DKK-1. ( D) qPCR of genes associated\n",
      "with cell activation in pericytes. ( E) Fluorescence images of αSMA showing\n",
      "the cytoskeleton of primary pericytes in control or stimulated conditions for\n",
      "24 h. (Scale bars, 25 μm.) ( F) Western blots showing phosphorylated LRP-6\n",
      "levels in pericytes 10 min after activation with TGF- βor WNT3a in the\n",
      "presence of vehicle or DKK-1 ( Upper ), and after activation with CTGF ( Lower )\n",
      "(G) Thirty-cycle RT-PCR showing the effect of TGF- βor TGF- β+DKK-1 on\n",
      "WNT ligand expression at 8 h. ( H) Fluorescence images and data quantifying\n",
      "nuclear GFP+(green) in TCF/Lef:H2B-GFPTrβcatenin reporter pericytes 16 h\n",
      "after stimulation with cytokines +/−DKK-1. ( I) Western blot time course of\n",
      "phosphorylated forms of P42/P44, JNK, P38, LRP-6, and FAK in pericytes ac-\n",
      "tivated by TGF- βor TGF- β+DKK-1. ( J) Western blots of phosphorylated\n",
      "forms of P42/P44, JNK, P38, and FAK in pericytes activated by CTGF or CTGF +\n",
      "DKK-1. ( Kand L) Graphs showing the effect of DKK-1, the canonical WNT\n",
      "inhibitor XAV939, the TGF βR1 kinase inhibitor SB431542, the P42/P44 in-\n",
      "hibitor U0126, or the JNK inhibitor SP600125 on TGF- β–stimulated ( K)o r\n",
      "CTGF-stimulated ( L) migration in quiescent pericytes. ( M) Graph showing\n",
      "the effect of TGF- βon the migration of Ctnnb1ﬂ/ﬂpericytes expressing Cre\n",
      "recombinase vs. Ctnnb1ﬂ/ﬂpericytes expressing control protein GFP. ( N) West-\n",
      "ern blots of pericyte proteins immunoprecipitated by anti-TGF βR1 antibodies or\n",
      "control antibodies detecting pLRP-6 or TGF βR1. (O) Graph showing the effect\n",
      "of expression of LRP-6 (wild type) or dominant-negative forms of LRP-6, 5m,\n",
      "orΔCo n3 T 3 ﬁbroblast migration in response to TGF- βand DKK-1. * P<0.05,\n",
      "**P<0.01, *** P<0.01. Error bars indicate SEM. Experiments are from n=4–7\n",
      "per group. All blots are representative of three experiments.\n",
      "1444 |www.pnas.org/cgi/doi/10.1073/pnas.1211179110 Ren et al.plays a role in the underlying tendency to migrate. To test\n",
      "whether LRP-6 also associated with the TGF βR complex,\n",
      "TGFβR1 was immunoprecipitated, and pLRP-6 was coprecipi-\n",
      "tated in resting cells. This association increased upon active sig-\n",
      "naling (Fig. 5 N). To determine the role of LRP-6 in TGF-\n",
      "β–stimulated migration, we studied expression of the dominant-\n",
      "negative LRP-6 mutants in ﬁbroblasts, as described above (Fig.\n",
      "4L). Strikingly, although TGF- β–mediated migration occurred in\n",
      "the presence of mutant LRP-6, DKK-1 was now ineffective (Fig.\n",
      "5O). Therefore, DKK-1 inhibits TGF- βmigration by an LRP-6 –\n",
      "dependent mechanism, and active LRP-6 can be detected in\n",
      "a signaling complex with TGF βR1. That TGF- βstimulates mi-\n",
      "gration in the presence of dominant-negative LRP-6 but DKK-1\n",
      "no longer functions suggests that the endogenous LRP-6 still may\n",
      "be able to interact with TGF- β, whereas DKK-1 binds equally to\n",
      "the mutant forms; further studies will be required to understand\n",
      "LRP-6 interactions with TGF βR1. Finally, to test the role of WNTs\n",
      "in TGF- β–mediated migration and in DKK-1 inhibition, we\n",
      "blocked WNT secretion using IWP2 as described above ( Fig. S7 ).\n",
      "In this setting TGF- β–stimulated migration was blocked com-\n",
      "pletely, suggesting WNT engagement of LRP-6 at the cell surface\n",
      "is necessary for TGF- βresponse in pericytes.\n",
      "These studies identify LRP-6 and signaling pathways down-\n",
      "stream of LRP-6 as therapeutic targets to intervene in ﬁbro-\n",
      "genesis of the kidney. In addition, they identify the soluble pro-\n",
      "tein DKK-1 and its receptor LRP-6 as important cofactors in\n",
      "multiple signaling pathways activated by TGF- β, CTGF, and\n",
      "PDGF ( Fig. S8 E). Because these multiple pathways are known to\n",
      "contribute to the development of ﬁbrosis and its longer-term\n",
      "consequences on organ function, therapeutically delivered DKK-\n",
      "1 or therapeutic targeting of the LRP-6 receptor are attractive\n",
      "strategies for treating ﬁbrosis, microvascular in ﬂammation, tubule\n",
      "injury, and microvascular rarefaction. Recent studies indicate that\n",
      "mesenchyme-derived cells, either pericytes or resident ﬁbroblasts,\n",
      "are present in all organs and tissues and are the primary source of\n",
      "scar-forming myo ﬁbroblasts in multiple tissues (2, 20). Therefore\n",
      "it is likely that the DKK-1/LRP-6 signaling pathway regulates\n",
      "ﬁbrogenesis in multiple organs and tissues.\n",
      "The MAPK signaling pathways from TGF βR signaling, CTGF\n",
      "signaling, and PDGFR signaling appear to be functionally critical\n",
      "in the mesenchyme-derived perivascular cells in kidney. These\n",
      "pathways have received relatively little attention, and earlier\n",
      "studies have assumed that the canonical pathways activated by\n",
      "these receptors have been dominant in the process of ﬁbrogenesis\n",
      "(14, 21); further studies should focus on these pathways in peri-\n",
      "cytes and ﬁbroblasts. Our studies show that LRP-6 (and poten-\n",
      "tially LRP-5), when bound by WNT ligands ,interacts physically\n",
      "with PDGFR βand TGF βR1 upon cognate ligand interaction\n",
      "(Fig. S8 E). It is possible, therefore, that LRP-6 is responsible fortransducing the signal that activates the MAPK pathways. Fur-\n",
      "ther studies should address this question. The fact that LRP-6 can\n",
      "function as a coreceptor for multiple pathways raises the question\n",
      "whether it requires an extracellular ligand, distinct from WNT\n",
      "ligands, to effect this interaction. It is likely that soluble ligands\n",
      "such as TGF- βand PDGF are presented at the cell surface bound\n",
      "to extracellular modulators and that these complexes may bind to\n",
      "and be responsible for the recruitment of LRP-6. The studies also\n",
      "indicate that in pericytes and myo ﬁbroblasts CTGF activates cells\n",
      "independently of TGF- β, whereas previous reports have sug-\n",
      "gested CTGF augments TGF- βresponses in ﬁbrogenesis. These\n",
      "studies suggest CTGF has a distinct receptor on pericytes and\n",
      "myoﬁbroblasts that has not been fully appreciated. Further\n",
      "studies should de ﬁne the CTGF receptor in these cells and the\n",
      "interaction of LRP-6 with that receptor.\n",
      "Genetic studies of LRP-6 and LRP-5 function in mice point to\n",
      "their major role in WNT pathway signal transduction in early and\n",
      "later embryogenesis (5). However, the initial studies suggested\n",
      "that LRP-5/LRP-6 mutations interfered with other signaling\n",
      "pathways, including FGFR1 and Notch Delta (22). Recently,\n",
      "LRP-6 mutations have been found not only in humans with re-\n",
      "duced bone mineral density but also in patients with premature\n",
      "cardiovascular disease (4, 5). These reports also have suggested\n",
      "a potential link between LRP-6 and PDGFR signaling. In the\n",
      "context of our current ﬁndings, these studies suggest that LRP-6\n",
      "may regulate multiple signaling pathways in mesenchyme-de-\n",
      "rived cells that play critical roles in bone mineralization, arterial\n",
      "wall functions, and microvascular wall functions.\n",
      "We conclude that DKK-1 is anti ﬁbrotic and anti-in ﬂammatory\n",
      "by inhibiting multiple signaling pathways in pericytes and myo ﬁ-\n",
      "broblasts through binding to the cell-surface receptors LRP-5/-6,\n",
      "which in turn act as coreceptors for multiple signaling pathways.\n",
      "Materials and Methods\n",
      "Mouse Models. Col1a1-GFPTr(Coll-GFP ),Axin2+/acz, and TCF/Lef:H2B-GFPTr\n",
      "mice were generated and maintained, and genotyping was performed as\n",
      "described (11, 12). Ctnn1ﬂ/ﬂmice were obtained from Jackson Labs. All studies\n",
      "were conducted in accordance with protocols approved by the Institutional\n",
      "Animal Care and Use Committee at the University of Washington.\n",
      "Statistical Analysis. Performed using GraphPad Prism (GraphPad Software).\n",
      "Signiﬁcance was evaluated by one-way ANOVA.\n",
      "ACKNOWLEDGMENTS. We thank Dr. Xi He, Dr. Roel Goldschmeding,\n",
      "Dr. William B. Stallcup, and Dr. Gustave Matute-Bello for advice, discussions,\n",
      "and sharing reagents. These studies were funded by National Institutes\n",
      "of Health Grants DK84077 and DK87389 (to J.S.D.), by the University of\n",
      "Washington, and by the University of Washington Institute for Stem Cell\n",
      "and Regenerative Medicine. A.K. was supported by the March of Dimes.\n",
      "1. Lupher ML, Jr., Gallatin WM (2006) Regulation of ﬁbrosis by the immune system. Adv\n",
      "Immunol 89:245 –288.\n",
      "2. Humphreys BD, et al. (2010) Fate tracing reveals the pericyte and not epithelial origin\n",
      "of myo ﬁbroblasts in kidney ﬁbrosis. Am J Pathol 176(1):85 –97.\n",
      "3. Lin SL, et al. (2011) Targeting endothelium-pericyte cross talk by inhibiting VEGF re-\n",
      "ceptor signaling attenuates kidney microvascular rarefaction and ﬁbrosis. Am J Pathol\n",
      "178(2):911 –923.\n",
      "4. Keramati AR, et al. (2011) Wild-type LRP6 inhibits, whereas atherosclerosis-linked\n",
      "LRP6R611C increases PDGF-dependent vascular smooth muscle cell proliferation. Proc\n",
      "Natl Acad Sci USA 108(5):1914 –1918.\n",
      "5. Tamai K, et al. (2000) LDL-receptor-relat ed proteins in Wnt signal transduction.\n",
      "Nature 407(6803):530 –535.\n",
      "6. Clevers H, Nusse R (2012) Wnt/ β-catenin signaling and disease. Cell149(6):1192 –1205.\n",
      "7. Poss KD (2010) Advances in understanding tissue regenerative capacity and mecha-\n",
      "nisms in animals. Nat Rev Genet 11(10):710 –722.\n",
      "8. Rodilla V, et al. (2009) Jagged1 is the pathological link between Wnt and Notch\n",
      "pathways in colorectal cancer. Proc Natl Acad Sci USA 106(15):6315 –6320.\n",
      "9. Lin SL, et al. (2010) Macrophage Wnt7b is critical for kidney repair and regeneration.\n",
      "Proc Natl Acad Sci USA 107(9):4194 –4199.\n",
      "10. Cheng JH, et al. (2008) Wnt antagonism inhibits hepatic stellate cell activation and\n",
      "liverﬁbrosis. Am J Physiol Gastrointest Liver Physiol 294(1):G39 –G49.\n",
      "11. Yu HM, et al. (2005) The role of Axin2 in calvarial morphogenesis and craniosynos-\n",
      "tosis. Development 132(8):1995 –2005.12. Ferrer-Vaquer A, et al. (2010) A sensitive and bright single-cell resolution live imaging\n",
      "reporter of Wnt/ß-catenin signaling in the mouse. BMC Dev Biol 10:121.\n",
      "13. Lin SL, Kisseleva T, Brenner DA, Duf ﬁeld JS (2008) Pericytes and perivascular ﬁbro-\n",
      "blasts are the primary source of collagen-producing cells in obstructive ﬁbrosis of the\n",
      "kidney. Am J Pathol 173(6):1617 –1627.\n",
      "14. Chen YT, et al. (2011) PDGF receptor signaling activates pericyte-myo ﬁbroblast\n",
      "transition in obstructive and post-ischemic kidney ﬁbrosis. Kidney Int 80:1170 –1181.\n",
      "15. Wang S, Denichilo M, Brubaker C, Hirschberg R (2001) Connective tissue growth\n",
      "factor in tubulointerstitial injury of diabetic nephropathy. Kidney Int 60(1):96 –105.\n",
      "16. ten Berge D, et al. (2011) Embryonic stem cells require Wnt proteins to prevent dif-\n",
      "ferentiation to epiblast stem cells. Nat Cell Biol 13(9):1070 –1075.\n",
      "17. Sieczkiewicz GJ, Herman IM (2003) TGF-beta 1 signaling controls retinal pericyte\n",
      "contractile protein expression. Microvasc Res 66(3):190 –196.\n",
      "18. Hall-Glenn F, et al. (2012) CCN2/CTGF is essential for pericyte adhesion and endo-\n",
      "thelial basement membrane formation during angiogenesis. PLoS ONE 7:e30562.\n",
      "19. Kawata K, et al. (2012) Role of low-density lipoprotein receptor related protein\n",
      "1 (LRP1) in CCN2 protein transport in chondrocytes. J Cell Sci 125:2965 –2972.\n",
      "20. Walker N, et al. (2011) Resident tissue-speci ﬁc mesenchymal progenitor cells con-\n",
      "tribute to ﬁbrogenesis in human lung allografts. Am J Pathol 178(6):2461 –2469.\n",
      "21. Leask A (2008) Targeting the TGFbeta, endothelin-1 and CCN2 axis to combat ﬁbrosis\n",
      "in scleroderma. Cell Signal 20(8):1409 –1414.\n",
      "22. Kokubu C, et al. (2004) Skeletal defects in ringelschwanz mutant mice reveal that\n",
      "Lrp6 is required for proper somitogenesis and osteogenesis. Development 131(21):\n",
      "5469 –5480.\n",
      "Ren et al. PNAS |January 22, 2013 |vol. 110 |no. 4 |1445\n",
      "MEDICAL SCIENCES\n",
      "View publication stats\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Tika parse trail\n",
    "parsed_pdf = parser.from_file('/Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/trial.txt')\n",
    "  \n",
    "# Printing of content \n",
    "print(parsed_pdf['content'])\n",
    "\n",
    "# print(type(data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "11b2ce8c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Text extracted from the second page and saved to /Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/extracted_texts/Katanin_Localization_Requires_Triplet_Microtubules.txt.\n",
      "Text extracted from the second page and saved to /Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/extracted_texts/LRP-6_is_co-receptor_for_multiple_fibrogenic_pathw.txt.\n",
      "Text extracted from the second page and saved to /Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/extracted_texts/pone.0092626.txt.\n",
      "Text extracted from the second page and saved to /Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/extracted_texts/miR-221_Promotes_Tumorigenesis_in_Human_Triple_Neg.txt.\n",
      "Text extracted from the second page and saved to /Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/extracted_texts/pone.0086342.txt.\n"
     ]
    }
   ],
   "source": [
    "# Folder path containing the PDF files\n",
    "pdf_folder = '/Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/trial dataset'\n",
    "\n",
    "# Output folder path to save extracted text files\n",
    "output_folder = '/Users/alberli/Desktop/USC/DSCI 550 DS at Scale/DSCI-550/hw2/extracted_texts'\n",
    "\n",
    "# Create the output folder if it doesn't exist\n",
    "os.makedirs(output_folder, exist_ok=True)\n",
    "\n",
    "# Loop through all PDF files in the folder\n",
    "for pdf_file in os.listdir(pdf_folder):\n",
    "    if pdf_file.endswith('.pdf'):\n",
    "        pdf_path = os.path.join(pdf_folder, pdf_file)\n",
    "        \n",
    "        # Create the output file name by replacing the '.pdf' extension with '.txt'\n",
    "        output_file_name = os.path.splitext(pdf_file)[0] + '.txt'\n",
    "        output_file_path = os.path.join(output_folder, output_file_name)\n",
    "        \n",
    "        # Extract text from the second page and save to the output file\n",
    "        extract_text_from_second_page(pdf_path, output_file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "47d7360c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'X-TIKA:Parsed-By': ['org.apache.tika.parser.DefaultParser', 'org.apache.tika.parser.csv.TextAndCSVParser'], 'X-TIKA:Parsed-By-Full-Set': ['org.apache.tika.parser.DefaultParser', 'org.apache.tika.parser.csv.TextAndCSVParser'], 'X-TIKA:content_handler': 'ToTextContentHandler', 'Content-Encoding': 'UTF-8', 'X-TIKA:parse_time_millis': '27', 'X-TIKA:embedded_depth': '0', 'resourceName': \"b'trial.txt'\", 'Content-Length': '44097', 'Content-Type': 'text/plain; charset=UTF-8'}\n"
     ]
    }
   ],
   "source": [
    "meta = parsed_pdf['metadata']\n",
    "print(meta)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "182cf0c0",
   "metadata": {},
   "source": [
    "### Format the tika parser with LaTex to differentiate title and the content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c0a578a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3a57c2a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "92a5aab7",
   "metadata": {},
   "source": [
    "## GPT-2 Model Part"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f37d7477",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import gpt_2_simple as gpt2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25eaf8b1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "c1a6be3c",
   "metadata": {},
   "source": [
    "## PDF via LaTex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1df6ce96",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ce9da857",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/latex": [
       "\\begin{twocolumn}\n",
       "Actual contents\n",
       "\\end{twocolumn}\n"
      ],
      "text/plain": [
       "<IPython.core.display.Latex object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%latex\n",
    "\\begin{twocolumn}\n",
    "Actual contents\n",
    "\\end{twocolumn}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e372384d",
   "metadata": {},
   "source": [
    "\\begin{center}\n",
    "    \\textbf{\\LARGE Title of the Document}\n",
    "\\end{center}\n",
    "\n",
    "\\begin{center}\n",
    "    \\textbf{\\large First Author Name} \\\\\n",
    "    \\textbf{\\large Second Author Name} \\\\\n",
    "\\end{center}\n",
    "\n",
    "\\begin{center}\n",
    "    Affiliation University Name Goes Here \\\\\n",
    "\\end{center}\n",
    "\n",
    "\\begin{twocolumn}\n",
    "    GPT Output Goes Here\n",
    "\\end{twocolumn}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "62515c1a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6    LPS Unmasking of Shigella flexneri Reveals Pre...\n",
       "7    miR-1915 and miR-1225-5p Regulate the Expressi...\n",
       "8    Mitochondrial Dysfunction Promotes Breast Canc...\n",
       "9    PrPST, a Soluble, Protease Resistant and Trunc...\n",
       "Name: Title, dtype: object"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bik['Title'][6:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "c9cbd73f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PDF file 'LPS_Unmasking_of_Shigella_flexneri_Reveals_Preferential_Localisation_of_Tagged_Outer_Membrane_Protease_IcsP_to_Septa_and_New_Poles.pdf' successfully generated!\n",
      "PDF file 'miR-1915_and_miR-1225-5p_Regulate_the_Expression_of_CD133,_PAX2_and_TLR2_in_Adult_Renal_Progenitor_Cells.pdf' successfully generated!\n",
      "PDF file 'Mitochondrial_Dysfunction_Promotes_Breast_Cancer_Cell_Migration_and_Invasion_through_HIF1a_Accumulation_via_Increased_Production_of_Reactive_Oxygen_Species.pdf' successfully generated!\n",
      "PDF file 'PrPST,_a_Soluble,_Protease_Resistant_and_Truncated_PrP_Form_Features_in_the_Pathogenesis_of_a_Genetic_Prion_Disease.pdf' successfully generated!\n"
     ]
    }
   ],
   "source": [
    "# Latex to pdf trial with title defined by the csv file to check format\n",
    "import subprocess\n",
    "\n",
    "# Define the LaTeX content as a string\n",
    "latex_content = r'''\n",
    "\\documentclass{article}  % Add this line to specify the document class\n",
    "\\begin{document}         % Add this line to start the document\n",
    "\n",
    "\\begin{center}\n",
    "    \\textbf{\\LARGE Title of the Document}\n",
    "\\end{center}\n",
    "\n",
    "\\begin{center}\n",
    "    \\textbf{\\large First Author Name} \\\\\n",
    "    \\textbf{\\large Second Author Name} \\\\\n",
    "\\end{center}\n",
    "\n",
    "\\begin{center}\n",
    "    Affiliation University Name Goes Here \\\\\n",
    "\\end{center}\n",
    "\n",
    "\\begin{twocolumn}\n",
    "The twoside option is not about placing text in a multicolumn fashion. \n",
    "While oneside is typical for articles and reports, \n",
    "the twoside option is loaded when you want to have a double-sided document, \n",
    "allowing you to set the margins, asymmetrically, in order to bind the document as a book.The twoside option is not about placing text in a multicolumn fashion. \n",
    "While oneside is typical for articles and reports, \n",
    "the twoside option is loaded when you want to have a double-sided document, \n",
    "allowing you to set the margins, asymmetrically, in order to bind the document as a book.The twoside option is not about placing text in a multicolumn fashion. \n",
    "While oneside is typical for articles and reports, \n",
    "the twoside option is loaded when you want to have a double-sided document, \n",
    "allowing you to set the margins, asymmetrically, in order to bind the document as a book.The twoside option is not about placing text in a multicolumn fashion. \n",
    "While oneside is typical for articles and reports, \n",
    "the twoside option is loaded when you want to have a double-sided document, \n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfaabcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "abcacbacbafbadfijaspofijasdfoiadfiasfa\n",
    "\n",
    "\\end{twocolumn}\n",
    "\n",
    "\\end{document}          \n",
    "'''\n",
    "# Loop over some of titles in the 'Title' column of the 'bik' dataset\n",
    "for title in bik['Title'][6:10]:\n",
    "    # Create a LaTeX file with the current title\n",
    "    input_file_name = f'{title.replace(\" \", \"_\")}.tex'\n",
    "    \n",
    "    # Save the LaTeX content to the .tex file\n",
    "    with open(input_file_name, 'w') as f:\n",
    "        f.write(latex_content)\n",
    "\n",
    "    # Full path to pdflatex\n",
    "    pdflatex_path = '/usr/local/bin/pdflatex'\n",
    "\n",
    "    # Run pdflatex command twice to ensure proper generation\n",
    "    for _ in range(2):\n",
    "        subprocess.run([pdflatex_path, '-jobname=' + os.path.splitext(input_file_name)[0], input_file_name],\n",
    "                       stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "\n",
    "\n",
    "    # Remove unnecessary files generated by pdflatex\n",
    "    os.remove(input_file_name)\n",
    "    for file_extension in ['log', 'aux']:\n",
    "        try:\n",
    "            subprocess.run(['rm', f'{os.path.splitext(input_file_name)[0]}.{file_extension}'])\n",
    "        except FileNotFoundError:\n",
    "            pass\n",
    "\n",
    "    # Check if the PDF file exists after running pdflatex twice\n",
    "    if os.path.exists(f'{os.path.splitext(input_file_name)[0]}.pdf'):\n",
    "        print(f\"PDF file '{os.path.splitext(input_file_name)[0]}.pdf' successfully generated!\")\n",
    "    else:\n",
    "        print(f\"PDF generation for '{title}' failed. Check for errors in the LaTeX code.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "3c92196d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PDF file 'Chronic_Morphine_Treatment_Attenuates_Cell_Growth_of_Human_BT474_Breast_Cancer_Cells_by_Rearrangement_of_the_ErbB_Signalling_Network.pdf' successfully generated!\n",
      "PDF generation for 'Katanin Localization Requires Triplet Microtubules in Chlamydomonas reinhardtii' failed. Check for errors in the LaTeX code.\n",
      "PDF file 'Curcumin_Modulates_the_Inflammatory_Response_and_Inhibits_Subsequent_Fibrosis_in_a_Mouse_Model_of_Viral-_induced_Acute_Respiratory_Distress_Syndrome.pdf' successfully generated!\n",
      "PDF file 'miR-221_Promotes_Tumorigenesis_in_Human_Triple_Negative_Breast_Cancer_Cells.pdf' successfully generated!\n",
      "PDF file 'HIV-1_Tat_Protein_Increases_Microglial_Outward_K+_Current_and_Resultant_Neurotoxic_Activity.pdf' successfully generated!\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "import os\n",
    "# Change the template to enable updating metadata of the pdf output according to \n",
    "# the bik dataset\n",
    "\n",
    "# Define the LaTeX template with placeholders\n",
    "latex_template = r'''\n",
    "\\documentclass{article}\n",
    "\\begin{document}\n",
    "\\begin{center}\n",
    "    \\textbf{\\LARGE %s}\n",
    "\\end{center}\n",
    "\n",
    "\\begin{center}\n",
    "    \\textbf{\\large %s} \\\\\n",
    "\\end{center}\n",
    "\n",
    "\\begin{center}\n",
    "    \\textbf %s \\\\\n",
    "\\end{center}\n",
    "\\begin{twocolumn}\n",
    "    GPT Outputs\n",
    "\\end{twocolumn}\n",
    "\\end{document}\n",
    "'''\n",
    "\n",
    "# Loop over the top 5 titles in the 'Title' column of the 'bik' dataset\n",
    "for idx, row in bik.head(5).iterrows():\n",
    "    title = row['Title']\n",
    "    Authors = row['Authors']\n",
    "    affiliation = row['university/institution']\n",
    "    \n",
    "    # Create a LaTeX file with the current title\n",
    "    input_file_name = f'{title.replace(\" \", \"_\")}.tex'\n",
    "    latex_content_with_data = latex_template % (title,Authors, affiliation)\n",
    "    \n",
    "    # Save the LaTeX content to the .tex file\n",
    "    with open(input_file_name, 'w') as f:\n",
    "        f.write(latex_content_with_data)\n",
    "\n",
    "    # Full path to pdflatex\n",
    "    pdflatex_path = '/usr/local/bin/pdflatex'\n",
    "\n",
    "    # Run pdflatex command twice to ensure proper generation\n",
    "    for _ in range(2):\n",
    "        subprocess.run([pdflatex_path, '-jobname=' + os.path.splitext(input_file_name)[0], input_file_name],\n",
    "                       stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)\n",
    "\n",
    "    # Remove the .tex file\n",
    "    os.remove(input_file_name)\n",
    "    for file_extension in ['log', 'aux']:\n",
    "        try:\n",
    "            subprocess.run(['rm', f'{os.path.splitext(input_file_name)[0]}.{file_extension}'])\n",
    "        except FileNotFoundError:\n",
    "            pass\n",
    "\n",
    "    # Check if the PDF file exists after running pdflatex twice\n",
    "    pdf_file_name = f'{os.path.splitext(input_file_name)[0]}.pdf'\n",
    "    if os.path.exists(pdf_file_name):\n",
    "        print(f\"PDF file '{pdf_file_name}' successfully generated!\")\n",
    "    else:\n",
    "        print(f\"PDF generation for '{title}' failed. Check for errors in the LaTeX code.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "febceab5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "aef61bfd",
   "metadata": {},
   "source": [
    "## Attempts to make the Tika extraction cleaner"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "a095109f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_empty_lines_and_add_new_paper(file_path):\n",
    "    # Read the content of the file\n",
    "    with open(file_path, 'r', encoding='utf-8') as file:\n",
    "        lines = file.readlines()\n",
    "\n",
    "    # Function to check if a line is empty\n",
    "    def is_empty_line(line):\n",
    "        return line.strip() == ''\n",
    "\n",
    "    # Initialize variables\n",
    "    empty_line_count = 0\n",
    "\n",
    "    for i, line in enumerate(lines):\n",
    "        if is_empty_line(line):\n",
    "            empty_line_count += 1\n",
    "            if empty_line_count > 30:\n",
    "                # Insert the \"New Paper\" line at the beginning of the block\n",
    "                lines.insert(i - empty_line_count + 1, '\\\\textbf{New Paper}\\n')\n",
    "                # Reset the counter\n",
    "                empty_line_count = 0\n",
    "        else:\n",
    "            # Reset the counter when a non-empty line is encountered\n",
    "            empty_line_count = 0\n",
    "\n",
    "    return lines\n",
    "\n",
    "# Usage example:\n",
    "file_path = 'result_USC.txt'\n",
    "modified_lines = count_empty_lines_and_add_new_paper(file_path)\n",
    "\n",
    "# Save the modified lines back to the file\n",
    "with open('trial1.txt', 'w', encoding='utf-8') as output_file:\n",
    "    output_file.write(''.join(modified_lines))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "9be0c6dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "# Function to remove lines containing any of the specified substrings as separate words\n",
    "def remove_lines_with_substrings(text_lines, substrings_to_remove):\n",
    "    cleaned_lines = [line for line in text_lines if not any(re.search(r'\\b{}\\b|{}'.format(substring, substring), line) for substring in substrings_to_remove)]\n",
    "    return cleaned_lines\n",
    "\n",
    "# Read the content of the file\n",
    "with open('trial1.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Substrings to remove\n",
    "substrings_to_remove = [\"Hindawi\",\"Volume\", \n",
    "                        \"http\", \"https\",\".com\",\".edu\",\".gov\", \n",
    "                        \"DOI\", \"doi\",\"Doi\", \n",
    "                        \"PLOS ONE\",\"Epub\",\"pone\",\"Evidence-Based Complementary and Alternative Medicine\",\n",
    "                       \"journal of\", \"cytokine\",\"pgen\",\"ppat\",\"infection and immunity\",\n",
    "                       \"international journal of oncology\"]\n",
    "\n",
    "# Remove lines containing any of the specified substrings from the content\n",
    "cleaned_lines = remove_lines_with_substrings(lines, substrings_to_remove)\n",
    "\n",
    "# Join the lines back into a single string\n",
    "cleaned_text = ''.join(cleaned_lines)\n",
    "\n",
    "# Save the cleaned text into a new file named \"trial2.txt\"\n",
    "with open('trial2.txt', 'w', encoding='utf-8') as output_file:\n",
    "    output_file.write(cleaned_text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "210f2e55",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "# Function to remove lines containing any of the specified substrings as separate words\n",
    "def remove_lines_with_substrings(text_lines, substrings_to_remove):\n",
    "    cleaned_lines = [line for line in text_lines if not any(re.search(r'\\b{}\\b'.format(re.escape(substring)), line) for substring in substrings_to_remove)]\n",
    "    return cleaned_lines\n",
    "\n",
    "# Read the content of the file\n",
    "with open('trial1.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Substrings to remove\n",
    "substrings_to_remove = [\"Hindawi Publishing Corporation\",\"Volume\", \n",
    "                        \"http\", \"https\",\".com\",\".edu\",\".gov\", \n",
    "                        \"DOI\", \"doi\",\"Doi\", \n",
    "                        \"PLOS ONE\",\"Epub\",\"pone\",\"Evidence-Based Complementary and Alternative Medicine\",\n",
    "                       \"journal of\", \"cytokine\",\"pgen\",\"ppat\",\"infection and immunity\",\n",
    "                       \"international journal of oncology\"]\n",
    "\n",
    "# Remove lines containing any of the specified substrings from the content\n",
    "cleaned_lines = remove_lines_with_substrings(lines, substrings_to_remove)\n",
    "\n",
    "# Join the lines back into a single string\n",
    "cleaned_text = ''.join(cleaned_lines)\n",
    "\n",
    "# Save the cleaned text into a new file named \"trial2.txt\"\n",
    "with open('trial2_5.txt', 'w', encoding='utf-8') as output_file:\n",
    "    output_file.write(cleaned_text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "092fb793",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to remove lines with only non-letter characters or \n",
    "# less than 3 letters\n",
    "def trimmer(text_lines):\n",
    "    cleaned_lines = [line for line in text_lines if \n",
    "                     any(char.isalpha() for char in line) and \n",
    "                     len([char for char in line if char.isalpha()]) >= 3]\n",
    "    return cleaned_lines\n",
    "\n",
    "# Read the content of the file\n",
    "with open('trial2_5.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Remove lines with only non-letter characters or less than 3 letters\n",
    "cleaned_lines = trimmer(lines)\n",
    "\n",
    "# Join the lines back into a single string\n",
    "cleaned_text = ''.join(cleaned_lines)\n",
    "\n",
    "# Save the cleaned text back to the \"trial2.txt\" file\n",
    "with open('trial3.txt', 'w', encoding='utf-8') as output_file:\n",
    "    output_file.write(cleaned_text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "6a06a021",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to adjust lines with consecutive \"New Paper\" phrases and add new lines above and below\n",
    "def line_adjustment(text_lines):\n",
    "    cleaned_lines = []\n",
    "    new_paper_count = 0\n",
    "    for line in text_lines:\n",
    "        if \"\\\\textbf{New Paper}\" in line:\n",
    "            new_paper_count += 1\n",
    "            if new_paper_count < 2:\n",
    "                cleaned_lines.append(\"\\n\" + line)\n",
    "        else:\n",
    "            new_paper_count = 0\n",
    "            cleaned_lines.append(line)\n",
    "    return cleaned_lines\n",
    "\n",
    "# Read the content of the file\n",
    "with open('trial3.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Remove lines with only non-letter characters or less than 3 letters\n",
    "cleaned_lines = trimmer(lines)\n",
    "\n",
    "# Adjust lines with consecutive \"New Paper\" phrases and add new lines above and below\n",
    "cleaned_lines = line_adjustment(cleaned_lines)\n",
    "\n",
    "# Join the lines back into a single string\n",
    "cleaned_text = ''.join(cleaned_lines)\n",
    "\n",
    "# Save the cleaned text back to the \"trial2.txt\" file\n",
    "with open('trial3.txt', 'w', encoding='utf-8') as output_file:\n",
    "    output_file.write(cleaned_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "f201530b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "# Function to remove lines containing any of the specified substrings as separate words\n",
    "def remove_lines_with_substrings(text_lines, substrings_to_remove):\n",
    "    cleaned_lines = [line for line in text_lines if not any(re.search(r'\\b{}\\b'.format(re.escape(substring)), line) \n",
    "                                                            for substring in substrings_to_remove)]\n",
    "    return cleaned_lines\n",
    "\n",
    "# Cleaner function to process the content after \"\\\\textbf{New Paper}\" phrases\n",
    "def cleaner_function(text_lines, substrings_to_remove_after_new_paper):\n",
    "    cleaned_lines = []\n",
    "    next_line = ''\n",
    "    inspect_flag = False\n",
    "\n",
    "    for line in text_lines:\n",
    "        if \"\\\\textbf{New Paper}\" in line:\n",
    "            inspect_flag = True\n",
    "            cleaned_lines.append(line)\n",
    "            continue\n",
    "\n",
    "        if inspect_flag:\n",
    "            # Process the immediate next line after \"\\\\textbf{New Paper}\"\n",
    "            if line.strip() == '':\n",
    "                continue\n",
    "            else:\n",
    "                next_line = line\n",
    "                cleaned_lines.extend(remove_lines_with_substrings([next_line], substrings_to_remove_after_new_paper))\n",
    "            inspect_flag = False\n",
    "        else:\n",
    "            cleaned_lines.append(line)\n",
    "\n",
    "    return cleaned_lines\n",
    "\n",
    "# Read the content of the file\n",
    "with open('trial3.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Substrings to remove after the \"\\\\textbf{New Paper}\" line\n",
    "# based on the excel output above\n",
    "substrings_to_remove_after_new_paper = [\"research\", \"reload\", \n",
    "                                        \"Retraction\", \"biomed\", \n",
    "                                        \"cytokine\", \"journal\", \n",
    "                                        \"deb\", \"pbio\"]\n",
    "\n",
    "# Clean the content after the \"\\\\textbf{New Paper}\" phrases\n",
    "cleaned_lines = cleaner_function(lines, substrings_to_remove_after_new_paper)\n",
    "\n",
    "# Join the lines back into a single string\n",
    "cleaned_text = ''.join(cleaned_lines)\n",
    "\n",
    "# Save the cleaned text into a new file named \"trial4.txt\"\n",
    "with open('trial4.txt', 'w', encoding='utf-8') as output_file:\n",
    "    output_file.write(cleaned_text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "f4e4c0ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check the first line after the New Paper line again, to see if\n",
    "# there most of them belongs to actual papers\n",
    "# Read the content of the file\n",
    "with open('trial4.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Generate the summary table and ranking\n",
    "summary_df, ranking_df = generate_summary_table(lines)\n",
    "\n",
    "# Save the summary dataframes into an Excel file named \"summary.xlsx\"\n",
    "with pd.ExcelWriter('summary2.xlsx', engine='xlsxwriter') as writer:\n",
    "    summary_df.to_excel(writer, sheet_name='Summary Table', index=False)\n",
    "    ranking_df.to_excel(writer, sheet_name='Summary Ranking', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "ef81931c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "\n",
    "# Read the content of the file\n",
    "with open('trial4.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Substrings to remove after the \"\\\\textbf{New Paper}\" line\n",
    "# based on the excel output above\n",
    "substrings_to_remove_after_new_paper = [\"research\", \"Research\",\"reload\", \n",
    "                                        \"Retraction\", \"biomed\", \n",
    "                                        \"cytokine\", \"journal\", \n",
    "                                        \"deb\", \"pbio\"]\n",
    "\n",
    "# Clean the content after the \"\\\\textbf{New Paper}\" phrases using the cleaner_function\n",
    "cleaned_lines = cleaner_function(lines, substrings_to_remove_after_new_paper)\n",
    "\n",
    "# Join the lines back into a single string\n",
    "cleaned_text = ''.join(cleaned_lines)\n",
    "\n",
    "# Save the cleaned text into a new file named \"trial4_5.txt\"\n",
    "with open('trial4_5.txt', 'w', encoding='utf-8') as output_file:\n",
    "    output_file.write(cleaned_text)\n",
    "\n",
    "# Check the first line after the New Paper line again, to see if\n",
    "# there most of them belongs to actual papers\n",
    "# Read the content of the file\n",
    "with open('trial4_5.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Generate the summary table and ranking\n",
    "summary_df, ranking_df = generate_summary_table(lines)\n",
    "\n",
    "# Save the summary dataframes into an Excel file named \"summary.xlsx\"\n",
    "with pd.ExcelWriter('summary2_5.xlsx', engine='xlsxwriter') as writer:\n",
    "    summary_df.to_excel(writer, sheet_name='Summary Table', index=False)\n",
    "    ranking_df.to_excel(writer, sheet_name='Summary Ranking', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8cc99e20",
   "metadata": {},
   "source": [
    "### Now after many times of cleaning, the three lines under the New Paper is highly likely to be the title itself. Thus, We'll make the necessary latex format based on it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "f704acfc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def final_latex(text_lines, substrings_to_remove_after_new_paper):\n",
    "    cleaned_lines = []\n",
    "    new_paper_count = 0\n",
    "\n",
    "    for line in text_lines:\n",
    "        if \"\\\\textbf{New Paper}\" in line:\n",
    "            new_paper_count += 1\n",
    "            if new_paper_count > 1:\n",
    "                cleaned_lines.append(\"\\\\end{document}\\n\")  # Close the previous content section\n",
    "            cleaned_lines.append(\"\\n\\\\documentclass{article}\\n\\\\begin{document}\\n\")\n",
    "            cleaned_lines.append(\"\\\\section*{}\\n\")\n",
    "        cleaned_lines.append(line)\n",
    "\n",
    "    # Add the closing content section for the last paper\n",
    "    if new_paper_count > 1:\n",
    "        cleaned_lines.append(\"\\\\end{document}\\n\")\n",
    "\n",
    "    return cleaned_lines\n",
    "\n",
    "# Rest of the code remains the same...\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "4e8614aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the content of the \"trial4_5.txt\" file\n",
    "with open('trial4_5.txt', 'r', encoding='utf-8') as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Substrings to remove after the \"\\\\textbf{New Paper}\" line\n",
    "# based on the excel output above\n",
    "substrings_to_remove_after_new_paper = [\"research\", \"reload\", \n",
    "                                        \"Retraction\", \"biomed\", \n",
    "                                        \"cytokine\", \"journal\", \n",
    "                                        \"deb\", \"pbio\"]\n",
    "\n",
    "# Clean the content and apply LaTeX formatting\n",
    "cleaned_lines = final_latex(lines, substrings_to_remove_after_new_paper)\n",
    "\n",
    "# Join the lines back into a single string\n",
    "cleaned_text = ''.join(cleaned_lines)\n",
    "\n",
    "# Save the cleaned text into a new file named \"final_output.tex\"\n",
    "with open('trial5.txt', 'w', encoding='utf-8') as output_file:\n",
    "    output_file.write(cleaned_text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "ef0a4869",
   "metadata": {},
   "outputs": [],
   "source": [
    "def content_function(input_file, output_file):\n",
    "    with open(input_file, 'r') as infile, open(output_file, 'w') as outfile:\n",
    "        add_section = False\n",
    "        count = 0\n",
    "        for line in infile:\n",
    "            outfile.write(line)\n",
    "            if '\\\\textbf{New Paper}' in line:\n",
    "                add_section = True\n",
    "                count = 0\n",
    "            if add_section:\n",
    "                count += 1\n",
    "                if count == 3:\n",
    "                    outfile.write('\\\\section*{}\\n')\n",
    "                    add_section = False\n",
    "                    count = 0\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    input_file = 'trial5.txt'\n",
    "    output_file = 'final_trial.txt'\n",
    "    content_function(input_file, output_file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "091c93d3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
